Sources/LLVM/MetadataType.swift

```diff
@@ -15,3 +15,185 @@ public struct MetadataType: IRType {
+  ///   - context: The context in which to create the metadata. Defaults to
+  ///              `nil`, which means the global context.
+  public init(values: [IRValue], in context: Context? = nil) {
+    var vals = values.map { $0.asLLVM() as Optional }
+    self.llvm = vals.withUnsafeMutableBufferPointer { buf in
+      if let context = context {
+        return LLVMMDNodeInContext(context.llvm, buf.baseAddress,
+                                   UInt32(buf.count))
+      } else {
+        return LLVMMDNode(buf.baseAddress, UInt32(buf.count))
+      }
+    }
+  }
+
+  /// Returns the underlying `LLVMValueRef` backing this node.
+  public func asLLVM() -> LLVMValueRef {
+    return llvm
+  }
+}
+
+/// A `MetadataString` is an arbitrary string  of metadata attached to an
+/// instruction or value.
+public struct MetadataString: IRMetadata, ExpressibleByStringLiteral {
+  let llvm: LLVMValueRef
+
+  /// Creates a `MetadataString` from an underlying LLVM value
+  ///
+  /// - parameter llvm: The LLVM value for this `MDString`
+  internal init(llvm: LLVMValueRef) {
+    self.llvm = llvm
+  }
+
+  /// Creates a `MetadataString` with the provided string as a value, in the
+  /// provided context. If no context is provided, it will be created in the
+  /// global context.
+  ///
+  /// - parameters:
+  ///   - string: The string with which to create this node.
+  ///   - context: The context in which to create this node.
+  public init(_ string: String, in context: Context? = nil) {
+    if let context = context {
+      self.llvm = LLVMMDStringInContext(context.llvm, string,
+                                        UInt32(string.utf8.count))
+    } else {
+      self.llvm = LLVMMDString(string, UInt32(string.utf8.count))
+    }
+  }
+
+  /// Creates a `MetadataString` from a `String` literal.
+  ///
+  /// - parameter value: The string with which to create this node.
+  public init(stringLiteral value: String) {
+    self.init(value)
+  }
+
+  /// Creates a `MetadataString` from a `UnicodeScalar` literal.
+  ///
+  /// - parameter value: The string with which to create this node.
+  public init(unicodeScalarLiteral value: String) {
+    self.init(value)
+  }
+
+  /// Creates a `MetadataString` from a `ExtendedGraphemeClusterLiteral`
+  /// literal.
+  ///
+  /// - parameter value: The string with which to create this node.
+  public init(extendedGraphemeClusterLiteral value: String) {
+    self.init(value)
+  }
+
+  /// Returns the underlying `LLVMValueRef` backing this string.
+  public func asLLVM() -> LLVMValueRef {
+    return llvm
+  }
+}
+
+/// Enumerates the possible kinds of metadata that LLVM supports.
+/// - note: These must match, exactly, the enum in `llvm/Metadata.h`
+public enum MetadataKind: UInt32 {
+  /// A simple metadata node with a `tuple` of operands.
+  case tuple                           =  0
+
+  /// A debug location in source code, used for debug info and otherwise.
+  case debugInfoLocation               =  1
```

Comment:
These aren't correct anyway -- it's a completely different enum.

I think there just isn't a way to generate `DIFile`/`DICompileUnit`/etc. from the C API. I'm gonna remove that enum and replace it with a matching one.